src/components/icons/ProgressIcon.tsx

```diff
@@ -3,7 +3,7 @@ import { css, styled } from "@storybook/theming";
 
 const { rotate360 } = animation;
 
-export const ProgressIcon = styled.div<{ onButton?: boolean }>(
+export const ProgressIcon = styled.div<{ onButton?: boolean | "IconButton" }>(
```

Comment:
1. We do. It may be used on a regular button, or on an IconButton.
2. I'll change the name. Let's go with `parentComponent: "Button" | "IconButton"` to avoid the overloaded type as well.